cmake/modules/soc.cmake

```diff
@@ -0,0 +1,68 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# Copyright (c) 2021, Nordic Semiconductor ASA
+
+# Configure SoC settings based on Kconfig settings and SoC root.
+#
+# This CMake module will set the following variables in the build system based
+# on Kconfig settings and selected SoC.
+#
+# If no implementation is available for the selected SoC an error will be raised.
+#
+# Outcome:
+# The following variables will be defined when this CMake module completes:
+#
+# - SOC_NAME:   Name of the SoC in use, identical to CONFIG_SOC
+# - SOC_SERIES: Name of the SoC series in use, identical to CONFIG_SOC_SERIES
+# - SOC_FAMILY: Name of the SoC family, identical to CONFIG_SOC_FAMILY
+# - SOC_PATH:   Path fragment defined by either SOC_NAME or SOC_FAMILY/SOC_SERIES.
+# - SOC_DIR:    Directory containing the SoC implementation
+#
+# Variable dependencies:
+# - SOC_ROOT: CMake list of SoC roots containing SoC implementations
+#
+# Cmake module dependencies:
+# - kconfig CMake module.
```

Comment:
Shouldn't this be included then?

Reply:
Also, this module uses ARCH, so we should

include(arch)

Reply:
ARCH needs to be documented as a dependency

Reply:
soc.cmake is an internal file so we can ensure ARCH is set before soc.cmake is included.